docker/docker/src/test/java/org/arquillian/cube/docker/impl/client/DockerImageControllerTest.java

```diff
@@ -0,0 +1,79 @@
+package org.arquillian.cube.docker.impl.client;
+
+
+import org.arquillian.cube.docker.impl.client.config.CubeContainer;
+import org.arquillian.cube.docker.impl.client.config.Image;
+import org.arquillian.cube.docker.impl.docker.DockerClientExecutor;
+import org.arquillian.cube.impl.model.LocalCubeRegistry;
+import org.arquillian.cube.spi.Cube;
+import org.arquillian.cube.spi.CubeConfiguration;
+import org.arquillian.cube.spi.CubeRegistry;
+import org.arquillian.cube.spi.event.lifecycle.AfterDestroy;
+import org.jboss.arquillian.core.api.annotation.ApplicationScoped;
+import org.jboss.arquillian.core.test.AbstractManagerTestBase;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DockerImageControllerTest extends AbstractManagerTestBase {
+
+    public static final String CUBE_ID = "pingpong";
+
+    @Mock
+    Cube cube;
+
+    @Mock
+    CubeContainer config;
+
+    @Override
+    protected void addExtensions(List<Class<?>> extensions) {
+        extensions.add(DockerImageController.class);
+        super.addExtensions(extensions);
+    }
+
+    @Before
+    public void before(){
```

Comment:
I would suggest avoiding this kind of usage of `@Before`. I would rather prefer to duplicate mocking setup in each and every test, so when I read such a test I instantly know what is going on. With before I not only have to remember it, but also jump to it when I start my code reading exercise from the test method itself.